### PR TITLE
Added Billy.config.allow_params whitelist feature

### PR DESCRIPTION
**Pull request by destinf**
Added the allow_params feature along with a Billy.config.use_ignore_params to
allow the user to select between using ignore_params to as a blacklist for
results and allow_params as a whitelist for results.

### CODE REVIEW

File: lib/billy/cache.rb [hunk 1]
```diff
@@ -70,7 +70,11 @@ def reset
     end
 
     def key(method, orig_url, body, log_key = false)
-      ignore_params = Billy.config.ignore_params.include?(format_url(orig_url, true))
+      if Billy.config.use_ignore_params == true
```
Comment (ronwsmith) [inline lib/billy/cache.rb]: ` == true` is not needed

File: lib/billy/cache.rb [hunk 2]
```diff
@@ -70,7 +70,11 @@ def reset
...
+        ignore_params = Billy.config.ignore_params.include?(format_url(orig_url, true))
+      else # Billy.config.use_ignore_params == false
```
Comment (ronwsmith) [inline lib/billy/cache.rb]: Please remove the comment here.

**Comment by destinf** [comment]
Hi @ronwsmith, I've made the changes as requested, please let me know if there's anything else that needs to be changed.

**Comment by ronwsmith** [comment]
Looks good, I'll try to cut a new release soon!

**Comment by ronwsmith** [comment]
Released in v0.11.0

**Pull request merged**